### PR TITLE
Enchanted generic response

### PR DESCRIPTION
**Pull request by immmus**
I thought that necessary add interfaces for our responses.
I made FFDCResponse and GenericResponse for more reusability of the code.
If have you questions, please ask them.

### CODE REVIEW

**Comment by immmus** [comment]
@planetf1 Hi I dont know that it is problem with gradle...

**Comment by planetf1** [comment]
> @planetf1 Hi I dont know that it is problem with gradle...

We needed the ffdc package as a dependency. I pushed the fix for you:-)
```
     implementation project(':open-metadata-implementation:common-services:ffdc-services')
```
We are part way through refactoring to a gradle build - for now this is doing compile only

However we need @davidradl to review the actual subject area change (will be next week)

**Comment by immmus** [comment]
@planetf1 Thank you:)

File: open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/GenericResponse.java
Comment (davidradl) [inline open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/GenericResponse.java:31]: I suggest not using getHead as the method name. I assume this refers to the head (first) element in the array, which is terminology referring to the implementation. I suggest changing this to getOnlyResult()- or similar-  which makes more sense for the caller.     
This led me to think that it might be clearer to have 2 child interfaces 
* GenericMultiResponse - with getResults method 
* GenericSingleResponse - with getResult method

The caller then uses the appropriate interface and associated method.

File: open-metadata-implementation/access-services/subject-area/subject-area-api/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/responses/SubjectAreaOMASAPIResponse.java
Comment (davidradl) [inline open-metadata-implementation/access-services/subject-area/subject-area-api/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/responses/SubjectAreaOMASAPIResponse.java:24]: I am not sure what `value = {"head"})` is doing here - I am not understanding; does this relate to the getHead methods?  Please could you add a comment to explain why it is needed and how it impacts the other parts of the code.

File: open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java
Comment (davidradl) [inline open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java:11]: Shouldn't the spelling of this be Parameterization

File: open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java
Comment (davidradl) [inline open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java:10]: should "enable capturing and passing a generic type" be "enable capturing and passing a generic result"

File: open-metadata-implementation/access-services/subject-area/subject-area-server/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/server/mappers/graph/LineTypeMapper.java
Comment (davidradl) [inline open-metadata-implementation/access-services/subject-area/subject-area-server/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/server/mappers/graph/LineTypeMapper.java:29]: I suggest changing this string to HasA so it matches the class name for consistency. Same for the other Lines.

File: open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/FFDCResponse.java
Comment (davidradl) [inline open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/FFDCResponse.java:31]: the comments say activity but the method name is Action. I suggest using the same noun to prevent potential confusion

**Review by davidradl — CHANGES_REQUESTED** [review]
I like what you have done here @immmus. I have suggested some minor improvements. I am not the component owner of the common services. You will need @mandy-chessell 's approval before it can be merged.

File: open-metadata-implementation/access-services/subject-area/subject-area-server/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/server/mappers/graph/LineTypeMapper.java
Comment (immmus) [reply open-metadata-implementation/access-services/subject-area/subject-area-server/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/server/mappers/graph/LineTypeMapper.java:29]: @davidradl I thought about this but I saw this docs https://egeria.odpi.org/open-metadata-publication/website/open-metadata-types/0380-Spine-Objects.html and I realized that in the repository the name is such

File: open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java
Comment (immmus) [reply open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java:11]: I this case it generic parameter is mark of result

File: open-metadata-implementation/access-services/subject-area/subject-area-api/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/responses/SubjectAreaOMASAPIResponse.java
Comment (immmus) [reply open-metadata-implementation/access-services/subject-area/subject-area-api/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/responses/SubjectAreaOMASAPIResponse.java:24]: Yes, `value = {"head"}`means that Jackson does not serialize result from getter (getHead). Because it does not need for clients in their json.

File: open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/GenericResponse.java
Comment (immmus) [reply open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/GenericResponse.java:31]: In this case, this is just an auxiliary method, for those cases when we clearly know that we have a single result in our response. I don't think that we should divide into 2 interfaces in this case, it's just an auxiliary method for more convenient work on the back end and nothing more.

File: open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java
Comment (immmus) [reply open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java:10]: Okay

File: open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/FFDCResponse.java
Comment (immmus) [reply open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/FFDCResponse.java:31]: Yes, okay I too think so

File: open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java
Comment (davidradl) [reply open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java:11]: the interface name is Parametrization, but this is not a proper word , the closest word is Parameterization. I wondered if that was what you intended.

File: open-metadata-implementation/access-services/subject-area/subject-area-server/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/server/mappers/graph/LineTypeMapper.java
Comment (davidradl) [reply open-metadata-implementation/access-services/subject-area/subject-area-server/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/server/mappers/graph/LineTypeMapper.java:29]: @immmus I would suggest the omas is consistent with its own concepts; it should not use field names and the like because they match omrs, if there is a more appropriate concept to expose in the omas. This is one ofthe value adds of the omas, it exposes a more appropriate concept to a persona.

File: open-metadata-implementation/access-services/subject-area/subject-area-api/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/responses/SubjectAreaOMASAPIResponse.java
Comment (davidradl) [reply open-metadata-implementation/access-services/subject-area/subject-area-api/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/responses/SubjectAreaOMASAPIResponse.java:24]: Ah ok thanks.

File: open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java
Comment (immmus) [reply open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java:11]: Ah okey, sorry for my English...

Comment (immmus) [reply open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java:11]: "In mathematics, and more specifically in geometry, parametrization (or parameterization; also parameterisation, parametrisation)". I read it in https://en.wikipedia.org/wiki/Parametrization

**Comment by planetf1** [comment]
The grade build is failing after the version bump. It is not due to this pr. I have now disabled the mandatory nature of the check and will look at resolving. . The best push or update should be fine. Apologies

**Comment by davidradl** [comment]
I would prefer to put in https://github.com/odpi/egeria/pull/3535 first, then merge to that master, as it touches some of the same files.

**Comment by davidradl** [comment]
@immmus I suggest merging in the base branch now the LineEnds  change has gone in and I will re-review

**Comment by immmus** [comment]
> @immmus I suggest merging in the base branch now the LineEnds change has gone in and I will re-review

We have some error after merge last changes. Error with fvt.

**Comment by immmus** [comment]
@davidradl There was an error in fvt after updating.

Screenshot 2020-08-29 at 01 05 00
Screenshot 2020-08-29 at 01 05 38

I resolve it in https://github.com/odpi/egeria/pull/3545

**Comment by immmus** [comment]
Hi @davidradl !
Please check and merge this ->  https://github.com/odpi/egeria/pull/3545 before current pr.

File: open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java
Comment (davidradl) [reply open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/Parametrization.java]: > Parametrization, also spelled parameterization, parametrisation or parameterisation, is the process of defining or choosing parameters. I read it in https://en.wikipedia.org/wiki/Parametrization

Nice evidence based response .

**Comment by davidradl** [comment]
> Hi @davidradl !
> Please check and merge this -> #3545 before current pr.

Sorry I just saw this comment - I have merged the other one

**Comment by davidradl** [comment]
> > @immmus I suggest merging in the base branch now the LineEnds change has gone in and I will re-review
> 
> We have some error after merge last changes. Error with fvt.

@immmus I am not sure how this could have happened, the FVT is run in the build and needs to succeed for the build to  allow the merge of any change. @planetf1  @immmus Any ideas why the build worked but the FVT had an NPE?

**Comment by immmus** [comment]
> > > @immmus I suggest merging in the base branch now the LineEnds change has gone in and I will re-review
> > 
> > 
> > We have some error after merge last changes. Error with fvt.
> 
> @immmus I am not sure how this could have happened, the FVT is run in the build and needs to succeed for the build to allow the merge of any change. @planetf1 @immmus Any ideas why the build worked but the FVT had an NPE?

I dont know why build succeed but the problem was in the code which I fixed in the #3545  pr.

**Comment by planetf1** [comment]
@davidradl the PR build should run the fvts as as you say should fail if the tests fail - indeed we've seen them fail.

What time build are you referring to that passed - we need to check the logs to see if we can see what happened. Perhaps it was an intermittent error.

The PR builds are at https://dev.azure.com/ODPi/Egeria/_build?definitionId=11&_a=summary

I see one PASSED today 48 mins ago
I also see two previous ones FAILED
Then one before that PASSED
.. in terms of this PR

I looked at the last failing one which clearly shows
```
2020-08-31T10:41:23.4391030Z [ERROR] Tests run: 18, Failures: 2, Errors: 0, Skipped: 0

```
and ends
```
2020-08-31T10:41:24.1533737Z [INFO] Subject Area OMAS FVT .............................. FAILURE [01:20 min]
2020-08-31T10:41:24.1534108Z [INFO] Open Metadata Open Types Test Resources ............ SKIPPED
2020-08-31T10:41:24.1534443Z [INFO] Open Types Test generator .......................... SKIPPED
2020-08-31T10:41:24.1538502Z [INFO] Open Types Test execution .......................... SKIPPED
2020-08-31T10:41:24.1539220Z [INFO] ------------------------------------------------------------------------
2020-08-31T10:41:24.1539505Z [INFO] BUILD FAILURE
```

Can you elaborate more on what's wrong with this, or what you saw that was odd?

**Comment by davidradl** [comment]
> Can you elaborate more on what's wrong with this, or what you saw that was odd?
@planetf1 I did not know about the errors until @immmus put in his fix, but this is something odd that the build failed after the  checkin was allowed. I am not sure what we can do to diagnose at this point, short of cloning the build and backing off to or before the failing code. I think it is something we need to keep an eye on

**Comment by immmus** [comment]
> > Can you elaborate more on what's wrong with this, or what you saw that was odd?
> > @planetf1 I did not know about the errors until @immmus put in his fix, but this is something odd that the build failed after the  checkin was allowed. I am not sure what we can do to diagnose at this point, short of cloning the build and backing off to or before the failing code. I think it is something we need to keep an eye on
> > @immmus , did you identified when the failing code was checked in so we can check the logs?


the crash happened when I pushed the latest changes from the master branch f809066

**Comment by planetf1** [comment]
I'm trying to understand the situation are describing - we need to be precise in terms of branches and merges.

My understanding from the text above, is that @immmus you noticed a build failure after you merged changes from MASTER into your branch (you said 'pushed the latest changes from the master branch' but I think this is what you meant?) Enchanted_generic_response - which is of course linked to this PR. That merge into your branch would have kicked off a new build which failed (in fact a few did) as the FVTs did indeed run and find an error.

I think the build that represented that merge is here -> https://dev.azure.com/ODPi/Egeria/_build/results?buildId=9986&view=logs&j=4dc6fbf2-95db-5f93-ea5a-117822a16d38&t=c86a1d55-7ff7-5cd4-1eec-934a3dda1e85
and reports 'HEAD is now at 958322a7ef Merge f80906668e6c2ef39898387e4600c6ce07a7849f into 2c4f9321b093e9339e0ae43eb4db4ed508ca9feb
'

That build failed in the FVT with
```
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.163 s - in org.odpi.openmetadata.accessservices.subjectarea.fvt.junit.CategoryHierarchyIT
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   RelationshipsIT.testRelationships:17 Unexpected exception thrown: java.lang.NullPointerException
[ERROR]   RelationshipsIT.testRelationships:17 Unexpected exception thrown: java.lang.NullPointerException
[INFO] 
[ERROR] Tests run: 18, Failures: 2, Errors: 0, Skipped: 0

```

This isn't unusual ... even if master is working just fine, merging back changes into a branch can cause build failures -- as the branch may need updating to fix something those master changes caused. 

I see no particular concern with the testing or build process here. Initially I read it as you saying that something was merged INTO master, and then caused the master code to be bad (and failing FVT). But this seems NOT to be the case (other than the fact I know you had a LOCAL issue @davidradl )

In other words, there's no protection about what anyone merges into their own branch - protection is only on the odpi master/release branches, to keep it safe & ensure tests run

Obviously the SMEs (@davidradl @immmus ) may be able to comment on exactly why the NPE occurred and pin down how the error was introduced into master, which may lead to us adding more tests if indeed it was a bug rather than just unfortunate.

If my understanding is incorrect please correct!

**Comment by davidradl** [comment]
> I'm trying to understand the situation are describing - we need to be precise in terms of branches and merges.
> 
> My understanding from the text above, is that @immmus you noticed a build failure after you merged changes from MASTER into your branch (you said 'pushed the latest changes from the master branch' but I think this is what you meant?) Enchanted_generic_response - which is of course linked to this PR. That merge into your branch would have kicked off a new build which failed (in fact a few did) as the FVTs did indeed run and find an error.
> 
> I think the build that represented that merge is here -> https://dev.azure.com/ODPi/Egeria/_build/results?buildId=9986&view=logs&j=4dc6fbf2-95db-5f93-ea5a-117822a16d38&t=c86a1d55-7ff7-5cd4-1eec-934a3dda1e85
> and reports 'HEAD is now at [958322a](https://github.com/odpi/egeria/commit/958322a7ef74e57f43c6e10304a940a6afb3f338) Merge [f809066](https://github.com/odpi/egeria/commit/f80906668e6c2ef39898387e4600c6ce07a7849f) into [2c4f932](https://github.com/odpi/egeria/commit/2c4f9321b093e9339e0ae43eb4db4ed508ca9feb)
> '
> 
> That build failed in the FVT with
> 
> ```
> [INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.163 s - in org.odpi.openmetadata.accessservices.subjectarea.fvt.junit.CategoryHierarchyIT
> [INFO] 
> [INFO] Results:
> [INFO] 
> [ERROR] Failures: 
> [ERROR]   RelationshipsIT.testRelationships:17 Unexpected exception thrown: java.lang.NullPointerException
> [ERROR]   RelationshipsIT.testRelationships:17 Unexpected exception thrown: java.lang.NullPointerException
> [INFO] 
> [ERROR] Tests run: 18, Failures: 2, Errors: 0, Skipped: 0
> ```
> 
> This isn't unusual ... even if master is working just fine, merging back changes into a branch can cause build failures -- as the branch may need updating to fix something those master changes caused.
> 
> I see no particular concern with the testing or build process here. Initially I read it as you saying that something was merged INTO master, and then caused the master code to be bad (and failing FVT). But this seems NOT to be the case (other than the fact I know you had a LOCAL issue @davidradl )
> 
> In other words, there's no protection about what anyone merges into their own branch - protection is only on the odpi master/release branches, to keep it safe & ensure tests run
> 
> Obviously the SMEs (@davidradl @immmus ) may be able to comment on exactly why the NPE occurred and pin down how the error was introduced into master, which may lead to us adding more tests if indeed it was a bug rather than just unfortunate.
> 
> I note we had a large open lineage merge in that period where your branch build failed, as well as another subject area refactor either of which could have broken or fixed your tests
> 
> If my understanding is incorrect please correct!

@planetf1 I agree. That makes sense; nice explanation.

**Comment by immmus** [comment]
Hi @davidradl !
When do you merge this pr?

**Comment by davidradl** [comment]
@immmus sorry I lost track of this one. I will look at it today or tomorrow.

File: open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/GenericResponse.java
Comment (davidradl) [inline open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/GenericResponse.java:8]: Please could you document this interface. I suggest something like. "If an OMAS uses Generic types to implement it's Java API, then the responses can implement this interface."

**Review by davidradl — COMMENTED** [review]
@immus I am going to test the UI tomorrow to check that the views still work after the change. I have made some suggestions on naming of variables.

File: open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/SubjectAreaNode.java
Comment (immmus) [reply open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/SubjectAreaNode.java:25]: @davidradl I sugget `DefaultSubjectAreaNodeClients`. I think it is better.

File: open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/SubjectAreaNodeClients.java
Comment (immmus) [reply open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/SubjectAreaNodeClients.java:20]: @davidradl These method is found in SubjectAreaNode"Clients" and return SubjectAreaNode"Client" that clearly implies return client so I dont want rename these methods with postfix "client".

File: open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/SubjectAreaNode.java
Comment (immmus) [reply open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/SubjectAreaNode.java:23]: @davidradl okay

File: open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/SubjectAreaNodeClients.java
Comment (davidradl) [reply open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/SubjectAreaNodeClients.java:20]: > @davidradl These method is found in SubjectAreaNode"Clients" and return SubjectAreaNode"Client" that clearly implies return client so I dont want rename these methods with postfix "client".

@immus I can see that, I read a method called glossaries to be returning glossaries; with this in mind you may want to change them. I will not be insistent on this if you do not want to, as I do not see it as that important.

File: open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/DefaultSubjectAreaNodeClients.java
Comment (davidradl) [inline open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/DefaultSubjectAreaNodeClients.java:53]: please use the standard way of doing messages for subject area -add a new entry into SubjectAreaErrorCode and refer to that

File: open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/DefaultSubjectAreaNodeClients.java
Comment (davidradl) [inline open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/DefaultSubjectAreaNodeClients.java:101]: please use the standard way of doing messages for subject area -add a new entry into SubjectAreaErrorCode and refer to that

File: open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/relationships/SubjectAreaLine.java
Comment (davidradl) [inline open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/relationships/SubjectAreaLine.java:68]: please use the standard way of doing messages for subject area -add a new entry into SubjectAreaErrorCode and refer to that

File: open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/relationships/SubjectAreaLine.java
Comment (davidradl) [inline open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/relationships/SubjectAreaLine.java:296]: please use the standard way of doing messages for subject area -add a new entry into SubjectAreaErrorCode and refer to that

File: open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/SubjectAreaNodeClient.java
Comment (davidradl) [inline open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/SubjectAreaNodeClient.java:12]: I am not clear why you call this a default. I would have thought that the scan would be standard and it picks up the content of the package using reflection. So we get classes loaded dynamically based on the scan. Are you envisaging a non-default scan - I am not sure what this would mean. Please let me know your thinking on the use of default - or remove it .

**Review by davidradl — COMMENTED** [review]
I cannot test the view side of this until the admin services are fixed in https://github.com/odpi/egeria/pull/3630. sorry for the time this has taken , I am trying to review it carefully, so I do not miss anything important.

File: open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/DefaultSubjectAreaNodeClients.java
Comment (immmus) [reply open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/DefaultSubjectAreaNodeClients.java:25]: It became possible to do it in java 5 because reflection and annotations appeared in the 5 version. =) So it is not later Java capabilities.
But I will try to comment it:)

File: open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/ResponseParameterization.java
Comment (immmus) [reply open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/ResponseParameterization.java:16]: It means that I know about it but it is not error.

File: open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/ResponseParameterization.java
Comment (immmus) [reply open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/ResponseParameterization.java:29]: It means that I know about it but it is not error.

File: open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/SubjectAreaNodeClient.java
Comment (immmus) [reply open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/nodes/SubjectAreaNodeClient.java:12]: Okay I remove this.

**Comment by immmus** [comment]
@davidradl Let's resolve these remaining conversations in a next pr?? Because this pr has been hanging for quite some time, I would like to get these changes for further work.

**Comment by davidradl** [comment]
@immus I tested the view services and they seem fine. Please could you address the review feedback I raised in a follow-on issue.

**Comment by immmus** [comment]
> @immus I tested the view services and they seem fine. Please could you address the review feedback I raised in a follow-on issue.

Thx for merge.
Can you give me link on issue?

**Pull request merged**
